packages/runtime/src/templates/edge-shared/utils.ts

```diff
@@ -31,12 +31,11 @@ export const addMiddlewareHeaders = async (
   // We need to await the response to get the origin headers, then we can add the ones from middleware.
   const res = await originResponse
   const response = new Response(res.body, res)
-  const originCookies = response.headers.get('set-cookie')
   middlewareResponse.headers.forEach((value, key) => {
-    response.headers.set(key, value)
-    // Append origin cookies after middleware cookies
-    if (key === 'set-cookie' && originCookies) {
-      response.headers.append(key, originCookies)
+    if (key === 'set-cookie') {
+      response.headers.append(key, value)
+    } else {
+      response.headers.set(key, value)
```

Comment:
Re-reading the Deno docs, will the `.append` just set a new header anyway if it doesn't exist so we don't need the `.set` in the `else` statement? They already have logic for checking if it's a `set-cookie` header too.

Reply:
Ah yes I thought of this, but I kept set in there as I think we may be potentially doing the `set` behaviour elsewhere (if a user is setting another type of header in their middleware) where we actually want it to work like `set` (not appending a second value to the header, as `append` will do with other types of headers)

Reply:
Ok thanks!